app/models/user.rb

```diff
@@ -77,7 +77,7 @@ class User < ActiveRecord::Base
           dependent: :destroy
   has_one :invited_user, dependent: :destroy
   has_one :user_notification_schedule, dependent: :destroy
-  has_many :passwords, class_name: "UserPassword", dependent: :destroy
+  has_many :passwords, class_name: "UserPassword", dependent: :destroy, autosave: true
```

Comment:
Wondering if we should have something like

```
has_many :expired_passwords, class_name: "UserPassword", dependent: :destroy

has_one :password, -> { where(expired: false) }, class_name: "UserEmail", dependent: :destroy, autosave: true
```

Then, we could do something along the lines of

```
def password=(value)
  record = self.password || self.build_password
  record.password = value
  record.salt = SecureRandom.hex(PASSWORD_SALT_LENGTH)
end
```

and we don't have to do any juggling of multiple `passwords` records, or handle expiring/destroying records.